forc-pkg/src/pkg.rs

```diff
@@ -1661,17 +1661,25 @@ pub fn check(
         let pkg = &plan.graph[node];
         let path = &plan.path_map[&pkg.id()];
         let manifest = ManifestFile::from_dir(path, sway_git_tag)?;
-        let ast_res = compile_ast(&manifest, &profile, dep_namespace)?;
-        if let CompileAstResult::Success { typed_program, .. } = &ast_res {
-            if let TreeType::Library { .. } = typed_program.kind.tree_type() {
-                namespace_map.insert(node, typed_program.root.namespace.clone());
+
+        let source = manifest.entry_string()?;
+        let sway_build_config =
+            sway_build_config(manifest.dir(), &manifest.entry_path(), &profile)?;
+        let parsed_result = sway_core::parse(source, Some(&sway_build_config));
+
+        if let Some(parse_program) = &parsed_result.value {
```

Comment:
If parsing has failed and this package is *not* the project package but is one of its dependencies, we should return an error. We cannot continue compilation if we cannot parse the dependency.

Currently, it looks like if a dependency fails to parse this will `panic!` on a following iteration when we attempt to retrieve the namespace for the package (which won't exist as we could not type check it).

Reply:
Good call, i'll make this change.